core/src/smartcontracts/wasm.rs

```diff
@@ -438,82 +476,141 @@ impl<'wrld> Runtime<'wrld> {
             query_judge,
         );
 
-        self.execute_with_state(account_id, bytes, state)
+        self.execute_with_state(bytes, state)
     }
 
-    /// Executes the given wasm smartcontract
+    /// Executes the given wasm trigger
     ///
     /// # Errors
     ///
     /// - if unable to construct wasm module or instance of wasm module
     /// - if unable to add fuel limit
     /// - if unable to find expected exports(main, memory, allocator)
+    /// - if unable to write data to the smart contract memory
     /// - if the execution of the smartcontract fails
-    pub fn execute(
+    pub fn execute_trigger(
         &mut self,
         wsv: &WorldStateView,
         account_id: &AccountId,
         bytes: impl AsRef<[u8]>,
+        event: Event,
     ) -> Result<(), Error> {
-        let state = State::new(wsv, account_id.clone(), self.config);
-        self.execute_with_state(account_id, bytes, state)
+        let state = State::new(wsv, account_id.clone(), self.config).with_triggering_event(event);
+        let mut store = self.create_store(state)?;
+        let smart_contract = self.create_smart_contract(&mut store, bytes)?;
+
+        let main_fn = smart_contract
+            .get_typed_func(&mut store, export::TRIGGER_MAIN_FN_NAME)
+            .map_err(Error::ExportNotFound)?;
+
+        // NOTE: This function takes ownership of the pointer
+        main_fn.call(&mut store, ()).map_err(Error::ExportFnCall)
```

Comment:
```suggestion
        self.execute_with_state(bytes, state)
```

Reply:
You are right, didn't mentioned that before